plugins/debuginfo-install.py

```diff
@@ -74,59 +75,133 @@ def configure(self):
+            # They can get there through globs and they break the installation
+            # of debug packages with the same version as the installed base
+            # packages. If the base package of a debuginfo package is not in
+            # the query, the user specified a debug package on the command
+            # line. We don't want to ignore those, so we will install them.
+            # But, in this case the version will not be matched to the
+            # installed version of the base package, as that would require
+            # another query and is further complicated if the user specifies a
+            # version themselves etc.
+            for name in list(package_dict.keys()):
+                if name.endswith(Package.DEBUGINFO_SUFFIX):
+                    if name[:-debuginfo_suffix_len] in package_dict:
+                        package_dict.pop(name)
+                if name.endswith(Package.DEBUGSOURCE_SUFFIX):
+                    if name[:-debugsource_suffix_len] in package_dict:
+                        package_dict.pop(name)
+
+            # attempt to install debuginfo and debugsource for the highest
+            # listed version of the package (in case the package is installed,
+            # only the installed version is listed)
+            for packages in package_dict.values():
+                if packages[0]._from_system:
+                    # for packages from system (installed) there can be more
+                    # packages with different architectures listed and we want
+                    # to install debuginfo for all of them
+
+                    # we need to split them by architectures and install the
+                    # latest version for each architecture
+                    arch_dict = {}
+
+                    for package in packages:
+                        arch_dict.setdefault(package.arch, []).append(package)
+
+                    for package_arch_list in arch_dict.values():
+                        package = package_arch_list[0]
+
+                        if not self._install_debug_from_system(package, package.debug_name):
+                            self.debuginfo_missing.add(str(package))
+
+                        if not self._install_debug_from_system(package, package.debugsource_name):
+                            self.debugsource_missing.add(str(package))
+
+                else:
+                    # if the first package is not from the system, we just want its name
+                    package = packages[0]
+                    if package.name.endswith(Package.DEBUGINFO_SUFFIX) \
+                            or package.name.endswith(Package.DEBUGSOURCE_SUFFIX):
+
+                        self._install((package, ))
+                        continue
+
+                    if solution["nevra"] is None:
+                        print("TODO We have this package and it may be the wrong one: %s", package)
```

Comment:
What about to use a similar construct like for installed packages?
```
                        if not self._install_debug_no_nevra(packages, package.debug_name):
                            # same problem with logging and relevant arch
                            self.debuginfo_missing.add(str(package))

                        if not self._install_debug_no_nevra(packages, package.debugsource_name):
                            self.debugsource_missing.add(str(package))
```
```
    def _install_debug_no_nevra(self, packages, debug_name):
        """The function is identical with proposed change of _install_debug_from_system"""
        nevras = []
        for pkg in packages:
            nevras.append("{}-{}:{}-{}.{}".format(
                debug_name, pkg.epoch, pkg.version, pkg.release, pkg.arch))
        query = self.base.sack.query().filterm(nevra_strict=nevras)
        if query:
            self._install(query)
            return True

        return False
```

Reply:
I've used this approach, just simplified a bit the `nevras` composition.